dataprofiler/profilers/unstructured_text_profile.py

```diff
@@ -118,20 +119,20 @@ def _merge_words(self, other, merged_profile):
         :type merged_profile: TextProfiler
         :return:
         """
-        if not self._is_case_sensitive:
-            merged_profile.word_count = self.word_count.copy()
-            additive_words = other.word_count
+        if self._is_case_sensitive == other._is_case_sensitive:
+            merged_profile.word_count = self.word_count + other.word_count
         else:
-            merged_profile.word_count = other.word_count.copy()
-            additive_words = self.word_count
-
-        for word in additive_words:
-            word_lower = word.lower()
-            if word_lower not in self._stop_words:
-                if self._is_case_sensitive:
-                    merged_profile.word_count[word] += additive_words[word]
-                else:
-                    merged_profile.word_count[word_lower] += additive_words[word]
+            if not self._is_case_sensitive:
+                lower_word_count = self.word_count
+                upper_word_count = other.word_count
+            else:
+                lower_word_count = other.word_count
+                upper_word_count = self.word_count
```

Comment:
Using "lower" and "upper" feels a bit strange here, since it isn't a matter of things being upper or lower cased, right? Or am I misunderstanding something here?

Reply:
This is also a separate issue, but should we allow merging of profiles if one is case sensitive and the other isn't? Because say in the old implementation, if one is case sensitive and the other is not, then words are still lowered when being put into the dictionary, potentially ignoring the case-sensitivity in the previous profile. Should we require they agree on case sensitivity and then lower or not based on what they agree on? Thoughts?

Reply:
I changed the var names, should look better now.